Sources/Defaults/Defaults.swift

```diff
@@ -292,13 +292,64 @@ extension Defaults {
 		}
 	}
 
-	// We still keep this as it can be useful to pass a dynamic array of keys.
 	/**
 	Observe updates to multiple stored values.
 
 	- Parameter keys: The keys to observe updates from.
 	- Parameter initial: Trigger an initial event on creation. This can be useful for setting default values on controls.
 
+	```swift
+	Task {
+		for await (foo, bar) in Defaults.updates([.foo, .bar]) {
+			print("Values changed:", foo, bar)
+		}
+	}
+	```
+	*/
+	public static func updates<each Value: Serializable>(
+		_ keys: repeat Key<each Value>,
+		initial: Bool = true
+	) -> AsyncStream<(repeat each Value)> {
+		.init { continuation in
+			func getCurrentValues() -> (repeat each Value) {
+				(repeat self[each keys])
+			}
+
+			var observations = [DefaultsObservation]()
+
+			if initial {
+				continuation.yield(getCurrentValues())
+			}
+
+			for key in repeat (each keys) {
+				let observation = DefaultsObservation(object: key.suite, key: key.name) { _, _  in
+					continuation.yield(getCurrentValues())
+				}
+
+				observation.start(options: [])
+				observations.append(observation)
+			}
+
+			let immutableObservations = observations
+
+			continuation.onTermination = { _ in
+				// `invalidate()` should be thread-safe, but it is not in practice.
```

Comment:
Curious about this. I thought that `invalidate()` would be thread-safe with Lock, 
Should we use `DispatchQueue.sync` instead inside the `DefaultsObservation`?

Reply:
I'm not 100% sure. I have seen KVO crashes in some of my apps that may have been caused by some threading problems in macOS. I prefer to keep it limited to this code until I can 100% confirm it. This is just a precaution.

Reply:
Got it! Thanks for the explanation!